zmq/src/caml_zmq_stubs.c

```diff
@@ -304,10 +305,11 @@ CAMLprim value caml_zmq_get_int64_option(value socket, value option_name) {
     CAMLreturn (caml_copy_int64(mark));
 }
 
-CAMLprim value caml_zmq_get_bytes_option(value socket, value option_name) {
-    CAMLparam2 (socket, option_name);
+CAMLprim value caml_zmq_get_bytes_option(value socket, value option_name, value option_maxlen) {
+    CAMLparam3 (socket, option_name, option_maxlen);
     char buffer[256];
-    size_t buffer_size = sizeof (buffer) - 1;
+    size_t buffer_size = Unsigned_long_val(option_maxlen);
+    assert(buffer_size <= sizeof (buffer));
```

Comment:
We should just copy the data returned from the zmq call, and not use `caml_copy_string`. Also there is an write pas end problem when adding the \0.